app/models/provider_interface/change_offer_form.rb

```diff
@@ -64,6 +75,15 @@ def self.course_option_matches_course?(record)
       record.course_option_id && CourseOption.find(record.course_option_id).course.id == record.course_id
     end
 
+    def self.study_mode_valid_for_course?(record)
+      course = Course.find(record.course_id) if record.course_id
+      course && (course.both_study_modes_available? || course.study_mode == record.study_mode)
```

Comment:
Does this feel like it wants to be a method on course: `course.supports_study_mode?(mode)` or equivalent?